src/ClientImpl.cpp

```diff
@@ -729,17 +734,22 @@ void ClientImpl::invoke(Procedure &proc, boost::shared_ptr<ProcedureCallback> ca
 
     if (m_outstandingRequests >= m_maxOutstandingRequests) {
     	if (m_listener.get() != NULL) {
+            m_backPressuredForOutstandingRequests = true;
             try {
                 m_listener->backpressure(true);
             } catch (const std::exception& e) {
                 std::cerr << "Exception thrown on invocation of backpressure callback: " << e.what() << std::endl;
             }
         }
     	// We are overloaded, we need to reject traffic and notify the caller
-        if (m_enableAbandon) {
+        if (m_enableAbandon && callback->honorDoAbandon()) {
             callback->abandon(ProcedureCallback::TOO_BUSY);
             return;
         }
+        else if (m_enableAbandon) {
+            // report to client the request was not abandoned
+            callback->abandon(ProcedureCallback::NOT_ABANDONED);
+        }
     }
```

Comment:
Do you need an else here where if m_outStanding has dropped and you indicated backpressure indicate backpressure->(false) and reset notification status.